### PR TITLE
userguide: expand documentation for rule profiling - v2

### PR DESCRIPTION
**Pull request by jufajardini**
The page about performance and rule profiling showed the table generated by rules profiling but didn't inform how to achieve nor find it.

Task #4359

Previous PR: https://github.com/OISF/suricata/pull/11440

Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/4359

Describe changes:
- fix typo
- expand abbreviation for clarity

### CODE REVIEW

File: doc/userguide/performance/rule-profiling.rst [hunk 1]
```diff
@@ -1,6 +1,14 @@
 Rule Profiling
 ==============
 
+Suricata can generate a rules performance report at the end of each session, if
+built with the ``enable-profiling`` option (see configuring :ref:`Rule profiling<rule-and-packet-profiling-settings>`).
+
+Rule profiling can also be enabled by building the engine with ``enable-profiling-rules``
```
Comment (catenacyber) [inline doc/userguide/performance/rule-profiling.rst:7]: What is the difference between the two ?

Comment (jufajardini) [reply doc/userguide/performance/rule-profiling.rst:7]: Good question. I think that `enable-profiling-rules` won't enable other profiling options, whereas `enable-profiling` will also generate stats for packet profiling. Not sure if there's more that our docs don't mention.

Comment (jufajardini) [reply doc/userguide/performance/rule-profiling.rst:7]: Should I try to improve this section to clarify the difference?

Comment (catenacyber) [reply doc/userguide/performance/rule-profiling.rst:7]: > Should I try to improve this section to clarify the difference?

I think so

**Comment by jufajardini** [comment]
Followed by: https://github.com/OISF/suricata/pull/11562

**Pull request closed without merging**